messages/helpers.go

```diff
@@ -183,6 +183,24 @@ func AllHaveLowerRound(messages []*proto.Message, round uint64) bool {
 	return true
 }
 
+// AllHaveSameRound checks if all messages have the same round
+func AllHaveSameRound(messages []*proto.Message) bool {
+	if len(messages) <= 1 {
+		return false
+	}
```

Comment:
Why are you enforcing the condition here that you must have at least `2` messages?

This is a utility method that is expected to return true even if the input set contains 1 element (since all elements of the input set would actually be of the same round)

I suggest you change this condition to:
```go
if len(messages) < 1 {
    return false
}
```

Reply:
Completely agree